dotCMS/src/main/java/com/dotcms/rendering/velocity/services/DotResourceLoader.java

```diff
@@ -0,0 +1,158 @@
+            final PageMode mode = PageMode.get(path[0]);
+            final String id1 = path[1].indexOf("_") > -1 ? path[1].substring(0, path[1].indexOf("_")) : path[1];
+            final String language = path[1].indexOf("_") > -1 ? path[1].substring(path[1].indexOf("_") + 1, path[1].length())
+                    : String.valueOf(APILocator.getLanguageAPI()
+                        .getDefaultLanguage()
+                        .getId());
+
+            final String id2 = path.length > 3 ? path[2] : null;
+
+
+            Logger.debug(this, "DotResourceLoader:\tInode: " + id1);
+
+            try {
+                switch (type) {
+                    case NOT_VELOCITY: {
+                        CacheLocator.getVeloctyResourceCache()
+                        .addMiss(path);
+                    throw new ResourceNotFoundException("Cannot find velocity file : " + path);
+                    }
+                    case CONTAINER: {
+                        return new ContainerLoader().writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case TEMPLATE: {
+
+                        return new TemplateLoader().writeObject(id1, id2, mode, language, filePath);
+
+                    }
+                    case CONTENT: {
+                        return new ContentletLoader().writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case FIELD: {
+                        return new FieldLoader().writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case CONTENT_TYPE: {
+                        return new ContentTypeLoader().writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case SITE: {
+                        return new SiteLoader().writeObject(id1, id2, mode, language, filePath);
+
+                    }
+                    case HTMLPAGE: {
+                        return new PageLoader().writeObject(id1, id2, mode, language, filePath);
+
+                    }
+                    case VELOCITY_MACROS: {
+                        return VTLLoader.instance()
+                            .writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case VTL: {
+                        return VTLLoader.instance()
+                            .writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case VELOCITY_LEGACY_VL: {
+                        return VTLLoader.instance()
+                            .writeObject(id1, id2, mode, language, filePath);
+                    }
+                    default: {
+                        CacheLocator.getVeloctyResourceCache()
+                            .addMiss(path);
+                        throw new ResourceNotFoundException("Cannot find velocity file : " + path);
+                    }
+                }
+
+            } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)